main.nf

```diff
@@ -56,6 +56,10 @@ vim: syntax=groovy
 
 version = '1.1'
 
+params.each{
+  if (!checkParams(it.toString().split('=')[0])) {exit 1, "params $it.toString().split('=')[0] is unknown, see --help for more information"}
```

Comment:
Can’t you move the full code into `checkParams`? As it is, `checkParams` really is `isAllowedParam`.